### PR TITLE
[5357] Duplicating a featured item in Editorial BP for the first time, causes issues with XB

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5357

### CODE REVIEW

File: ui/app/src/services/content.ts [hunk 1]
```diff
@@ -388,22 +388,76 @@ export function duplicateItem(
...
   path: string
 ): Observable<any> {
-  return performMutation(
-    site,
-    path,
-    (element) => {
+  return fetchContentDOM(site, path).pipe(
+    switchMap((doc) => {
+      const documentModelId = doc.querySelector(':scope > objectId').innerHTML.trim();
+      let elementToDuplicate = doc.documentElement;
```
Comment (rart) [inline ui/app/src/services/content.ts]: Rename appropriately

Comment (rart) [inline ui/app/src/services/content.ts]: Rename appropriately

File: ui/app/src/services/content.ts [hunk 2]
```diff
@@ -388,22 +388,76 @@ export function duplicateItem(
...
+          writeContentUrl({
+            site,
+            path: path,
+            unlock: 'true',
+            fileName: getInnerHtml(doc.querySelector(':scope > file-name'))
+          }),
+          serialize(doc)
+        ).pipe(mapTo({ updatedDocument: doc }));
+      } else {
+        const item: Element = extractNode(elementToDuplicate, fieldId, targetIndex).cloneNode(true) as Element;
```
Comment (rart) [inline ui/app/src/services/content.ts]: Duplicate line

Comment (rart) [inline ui/app/src/services/content.ts]: Duplicate line

File: ui/app/src/services/content.ts [hunk 3]
```diff
@@ -388,22 +388,76 @@ export function duplicateItem(
...
-      updateElementComponentsId(item);
-      field.appendChild(item);
-    },
-    modelId
+
+      if (isEmbedded) {
+        updateItemId(item);
+        updateElementComponentsId(item);
+        field.appendChild(item);
+        updateModifiedDateElement(elementToDuplicate);
```
Comment (rart) [inline ui/app/src/services/content.ts]: Check if this can be refactored above (common to both flows)

Comment (rart) [inline ui/app/src/services/content.ts]: Check if this can be refactored above (common to both flows)

**Review by rart — CHANGES_REQUESTED** [review]

**Comment by jvega190** [comment]
@rart I also updated the code to update embedded components ids inside the item being duplicated

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 4]
```diff
@@ -662,16 +662,18 @@ export function PreviewConcierge(props: PropsWithChildren<{}>) {
...
-              hostToGuest$.next(duplicateItemOperationComplete());
-              enqueueSnackbar(formatMessage(guestMessages.duplicateItemOperationComplete));
-            },
-            error() {
-              hostToGuest$.next(duplicateItemOperationFailed());
-              enqueueSnackbar(formatMessage(guestMessages.duplicateItemOperationFailed));
-            }
-          });
+          duplicateItem(siteId, modelId, fieldId, index, path)
+            .pipe(switchMap(({ newItem }) => fetchContentInstanceDescriptor(siteId, newItem.path, { flatten: true })))
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]: I think we need to do a little bit more here. See `issueDescriptorRequest`. Perhaps we can even invoke that one.

File: ui/guest/src/contentController.ts [hunk 5]
```diff
@@ -339,6 +341,22 @@ export function updateField(modelId: string, fieldId: string, index: string | nu
...
+    .subscribe({
+      next({ payload }) {
+        const modelId = payload.model.craftercms.id;
+        // Update the model cache
+        models$.next({
+          ...models,
+          [modelId]: payload.model
+        });
+      }
+    });
```
Comment (rart) [inline ui/guest/src/contentController.ts]: I think If we manage to use `issueDescriptorRequest` we won't need this.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/contentController.ts [hunk 6]
```diff
@@ -28,6 +28,7 @@ import {
   contentTypesResponse,
   deleteItemOperation,
   duplicateItemOperation,
+  duplicateItemOperationComplete,
```
Comment (rart) [inline ui/guest/src/contentController.ts]: Not in use

File: ui/guest/src/contentController.ts [hunk 7]
```diff
@@ -43,6 +44,7 @@ import { parseDescriptor, preParseSearchResults } from '@craftercms/content';
 import { crafterConf } from '@craftercms/classes';
 import { getDefaultValue } from '@craftercms/studio-ui/utils/contentType';
 import { ModelHierarchyDescriptor, ModelHierarchyMap, modelsToLookup } from '@craftercms/studio-ui/utils/content';
+import { message$ } from './utils/communicator';
```
Comment (rart) [inline ui/guest/src/contentController.ts]: Not in use

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**